### PR TITLE
[analyzer][NFC] Migrate loc::ConcreteInt to use APSIntPtr (3/4)

### PR DESCRIPTION
**Pull request by steakhal**

### CODE REVIEW

File: clang/include/clang/StaticAnalyzer/Core/PathSensitive/SVals.h [hunk 1]
```diff
@@ -514,9 +514,12 @@ class MemRegionVal : public Loc {
 
 class ConcreteInt : public Loc {
 public:
-  explicit ConcreteInt(const llvm::APSInt &V) : Loc(ConcreteIntKind, &V) {}
+  explicit ConcreteInt(APSIntPtr V) : Loc(ConcreteIntKind, V.get()) {}
```
Comment (Xazax-hun) [inline clang/include/clang/StaticAnalyzer/Core/PathSensitive/SVals.h:517]: Is it possible to store `APSIntPtr` in `Loc`?

Comment (steakhal) [reply clang/include/clang/StaticAnalyzer/Core/PathSensitive/SVals.h:517]: In short, I don't think. See https://github.com/llvm/llvm-project/pull/120436#discussion_r1890504894.

Comment (steakhal) [reply clang/include/clang/StaticAnalyzer/Core/PathSensitive/SVals.h:517]: I guess this one is resolved now. Is it?
Anything else blocking this PR?

**Review by Xazax-hun — APPROVED** [review]

**Comment by steakhal** [comment]
### Merge activity

* **Dec 19, 6:57 AM EST**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/llvm/llvm-project/120437).

**Pull request merged**